onnx/common/ir.h

```diff
@@ -767,6 +767,53 @@ struct Node : public Attributes<Node> {
   }
 };
 
+// A class with the same properties as OperatorSetIdProto, but without protobuf
+// overhead, resulting in a simpler and more readable workflow.
+class OpSetID final {
+  private:
+    std::string domain_;
+    int64_t version_;
+
+  public:
+    explicit OpSetID(const std::string& target) {
+      std::stringstream ss(target);
+      std::string segment;
+      std::vector<std::string> seglist;
+      while (std::getline(ss, segment, '$')) {
+        seglist.push_back(segment);
+      }
+      ONNX_ASSERTM(seglist.size() == 2, "string must be of format $<domain>$<version>")
+      domain_ = seglist[0];
+      version_ = std::stoi(seglist[1].c_str());
+    }
+
+    explicit OpSetID(const ONNX_NAMESPACE::OperatorSetIdProto& proto)
+      :domain_(proto.domain()), version_(proto.version()) {}
+
+    // Default Domain Constructor
+    explicit OpSetID(const int64_t& version)
+      :domain_(""), version_(version) {}
+
+    explicit OpSetID(const std::string& domain, int64_t version)
+      :domain_(domain), version_(version) {}
+
+    std::string toString() const {
+      return "$" + domain_ + "$" + std::to_string(version_);
+    }
+
+    const std::string& domain() const {
+      return domain_;
+    }
+
+    const int64_t& version() const {
```

Comment:
For primitive type, no need to use/return reference, instead, return value is even faster.